src/Driver/AbstractOracleDriver/EasyConnectString.php

```diff
@@ -70,14 +70,21 @@ public static function fromConnectionParameters(array $params): self
             $connectData['INSTANCE_NAME'] = $params['instancename'];
         }
 
+        $protocol = null;
+        if (isset($params['driverOptions'], $params['driverOptions']['protocol'])) {
+            if ($params['driverOptions']['protocol'] === 'TCPS') {
+                $protocol = 'TCPS';
+            }
+        }
```

Comment:
Why hard-code the allowed values? The logic here might be as simple as:
```php
$protocol = $params['driverOptions']['protocol'] ?? 'TCP';
```

Reply:
@morozov I guess I was trying to restrict the chance of someone using the wrong thing. If that's not something for me to worry about, I can do that.

Reply:
It's the job of the database or the driver to worry about that. Otherwise, DBAL may forbid the usage of potentially valid values.

Reply:
I have updated the code to use your suggestion, but I moved the default into the connect string directly